presto-docs/src/main/sphinx/optimizer/statistics.rst

```diff
@@ -0,0 +1,147 @@
+     nationkey   | NULL      |                  25.0 |            0.0 | NULL      | 0                  | 24
+     NULL        | NULL      | NULL                  | NULL           |      25.0 | NULL               | NULL
+     (5 rows)
+
+
+    presto:default> SHOW STATS FOR (SELECT * FROM nation WHERE nationkey > 10);
+
+     column_name | data_size | distinct_values_count | nulls_fraction | row_count |     low_value      |     high_value
+    -------------+-----------+-----------------------+----------------+-----------+--------------------+--------------------
+     regionkey   | NULL      |                   5.0 |            0.0 | NULL      | 0                  | 4
+     name        | NULL      |                   9.0 |            0.0 | NULL      | IRAN               | VIETNAM
+     comment     | NULL      |                  14.0 |            0.0 | NULL      |  pending excuse... | y final package...
+     nationkey   | NULL      |                   3.0 |            0.0 | NULL      | 10                 | 24
+     NULL        | NULL      | NULL                  | NULL           |      25.0 | NULL               | NULL
+     (5 rows)
+
+If provided ``SELECT`` will filter out all of the partitions (all table layouts),
+then the ``SHOW STATS`` will return no statistic which will be represented as in example below.
+
+.. code-block:: none
+
+    presto:default> SHOW STATS FOR (SELECT * FROM nation WHERE nationkey > 999);
+
+     column_name
+    -------------
+     NULL
+    (1 row)
+
+Note, that currently providing ``column_list`` instead of ``*`` in ``SELECT`` will not influence the output table.
+
+For example:
+
+.. code-block:: none
+
+    presto:default> SHOW STATS FOR (SELECT comment FROM nation WHERE nationkey > 10);
+
+     column_name | data_size | distinct_values_count | nulls_fraction | row_count |     low_value      |     high_value
+    -------------+-----------+-----------------------+----------------+-----------+--------------------+--------------------
+     regionkey   | NULL      |                   5.0 |            0.0 | NULL      | 0                  | 4
+     name        | NULL      |                   9.0 |            0.0 | NULL      | IRAN               | VIETNAM
+     comment     | NULL      |                  14.0 |            0.0 | NULL      |  pending excuse... | y final package...
+     nationkey   | NULL      |                   3.0 |            0.0 | NULL      | 10                 | 24
+     NULL        | NULL      | NULL                  | NULL           |      25.0 | NULL               | NULL
+     (5 rows)
+
+
+Updating Statistics For Hive Tables
+-----------------------------------
+
+For the Hive connector, Presto uses the statistics that are managed by Hive and exposed via the Hive metastore API.
+Depending on the Hive configuration, table statistics may not be updated automatically.
+
+If statistics are not updated automatically, the user needs to trigger a statistics update via the Hive CLI.
```

Comment:
Please update this section to use Presto's `ANALYZE`

Reply:
I am dropping this whole section (below for reference). I added ``For the Hive connector, refer to
:ref:`Hive connector <hive_analyze>` documentation if you want to update table
statistics.`` instead.

```
Updating Statistics For Hive Tables
-----------------------------------

For the Hive connector, Presto uses the statistics that are managed by Hive and
exposed via the Hive metastore API.  Depending on the Hive configuration, table
statistics may not be updated automatically.

If statistics are not updated automatically, the user needs to trigger a
statistics update via the Hive CLI.

The following command can be used in the Hive CLI to update table statistics for
non-partitioned table ``t``::

        hive> ANALYZE TABLE t COMPUTE STATISTICS;
        hive> ANALYZE TABLE t COMPUTE STATISTICS FOR COLUMNS;

For partitioned tables, partitioning information must be specified in the
command.  Assuming table ``t`` has two partitioning keys ``a`` and ``b``, the
following command would update the table statistics for all partitions::

        hive> ANALYZE TABLE t PARTITION (a, b) COMPUTE STATISTICS FOR COLUMNS;

It is also possible to update statistics for just a subset of partitions.  This
command will update statistics for all partitions for which partitioning key
``a`` is equal to ``1``::

        hive> ANALYZE TABLE t PARTITION (a=1, b) COMPUTE STATISTICS FOR COLUMNS;

And this command will update statistics for just one partition::

        hive> ANALYZE TABLE t PARTITION (a=1, b=5) COMPUTE STATISTICS FOR COLUMNS;

For documentation on Hive's statistics mechanism see
https://cwiki.apache.org/confluence/display/Hive/StatsDev
```

Reply:
@mattsfuller